test/integration/protocol_integration_test.cc

```diff
@@ -784,6 +784,29 @@ TEST_P(DownstreamProtocolIntegrationTest, MissingHeadersLocalReplyWithBodyBytesC
                                        BytesCountExpectation(7, 10, 7, 8));
 }
 
+TEST_P(DownstreamProtocolIntegrationTest, TeSanitization) {
+  autonomous_upstream_ = true;
+  config_helper_.addRuntimeOverride("envoy.reloadable_features.sanitize_te", "true");
+
+  if (downstreamProtocol() == Http::CodecType::HTTP1) {
+    default_request_headers_.setTE("gzip");
+  } else {
+    return;
+  }
```

Comment:
nit: Early return: 

```
if (downstreamProtocol() != Http::CodecType::HTTP1) {
  return;
}
default_request_headers_.setTE("gzip");
```

Optionally put this early return at the top of the test to make it clear that it's H1-specific.